### ISSUE TITLE
Extract parser as standalone module

### ISSUE THREAD

**Issue by wilsonpage** [mozilla-magnet/magnet-metadata-service#6]
I think it could be possible to have this library working in both Node an Browser. This is important if we are to discover local (mdns) services.

``` js
var parse = require('pw-ws/parser');
parse('http://twitter.com/wilsonpage').then(metadata => ...);
```

In order to do this we need to use a request library that works in both browser and node (eg. [`superagent`](https://github.com/visionmedia/superagent)) and abstract DOM parsing behind a module that uses jsdom in Node and real DOM in browser. Package.json's `"browser"` field can help with this.

**Comment by wilsonpage** [comment]
It's important that this happens sooner rather than later so that we don't have to unpick much node specific code.

**Comment by arcturus** [comment]
We are close, but not 100% there.

For data fetching we are using a library that is a polyfill for `fetch`, [node-fetch](https://www.npmjs.com/package/node-fetch).

For parsing we are using jsdom, so we are compatilbe with a window object.

What we need to do is package the current parser (everything under metadata) to be able to be loaded by node or browser. I suggest to move the metadata folder to a different package, and have build step to build for browser or node.

But, for a poc so far we are ok :P

**Comment by arcturus** [comment]
Is this fixed by #15 ?

**Comment by wilsonpage** [comment]
Yep
